samples/sensor/kx022/src/main.c

```diff
@@ -0,0 +1,428 @@
+	printf("Tilt state %ld\n", (tmp & KX022_MASK_CNTL1_TPE));
+
+	rc = kx022_read_register_value(dev, KX022_ODCNTL, &tmp);
+	printf("Accelerometer odr %ld\n", (tmp & KX022_MASK_ODCNTL_OSA));
+
+	rc = kx022_read_register_value(dev, KX022_INC1, &tmp);
+	printf("Pin 1 interrupt state %ld\n", (tmp & KX022_MASK_INC5_IEN2) >> BITWISE_SHIFT_5);
+	printf("Pin 1 polarity %ld\n", (tmp & KX022_MASK_INC5_IEA2) >> BITWISE_SHIFT_4);
+
+	rc = kx022_read_register_value(dev, KX022_INC4, &tmp);
+	printf("Pin 1 tap report state %ld\n", (tmp & KX022_MASK_INC4_TDTI1) >> BITWISE_SHIFT_2); 
+	printf("Pin 1 motion report state %ld\n", (tmp & KX022_MASK_INC4_WUFI1) >> BITWISE_SHIFT_1); 
+	printf("Pin 1 tilt report state %ld\n", (tmp & KX022_MASK_INC4_TPI1)); 
+}
+
+static void accel_motion_cfg(const struct device *dev)
+{
+	int rc;
+	uint8_t tmp;
+
+	rc = kx022_read_register_value(dev, KX022_INC2, &tmp);
+	printf("Motion detect xn direction %ld\n", (tmp & KX022_MASK_INC2_XNWUE) >> BITWISE_SHIFT_5);
+	printf("Motion detect xp direction %ld\n", (tmp & KX022_MASK_INC2_XPWUE) >> BITWISE_SHIFT_4);
+	printf("Motion detect yn direction %ld\n", (tmp & KX022_MASK_INC2_YNWUE) >> BITWISE_SHIFT_3);
+	printf("Motion detect yp direction %ld\n", (tmp & KX022_MASK_INC2_YPWUE) >> BITWISE_SHIFT_2);
+	printf("Motion detect zn direction %ld\n", (tmp & KX022_MASK_INC2_ZNWUE) >> BITWISE_SHIFT_1);
+	printf("Motion detect zp direction %ld\n", (tmp & KX022_MASK_INC2_ZPWUE));
+
+	rc = kx022_read_register_value(dev, KX022_CNTL3, &tmp);
+	printf("Motion detect odr %ld\n", (tmp & KX022_MASK_CNTL3_OWUF));
+
+	rc = kx022_read_register_value(dev, KX022_WUFC, &tmp);
+	printf("Motion detection timer %d\n", tmp);
+
+	rc = kx022_read_register_value(dev, KX022_ATH, &tmp);
+	printf("Motion detect wake up threshold %d\n", tmp);
+}
+
+static void accel_tilt_cfg(const struct device *dev)
+{
+	int rc;
+	uint8_t tmp;
+
+	rc = kx022_read_register_value(dev, KX022_CNTL2, &tmp);
+	printf("Tilt xn direction %ld\n", (tmp & KX022_MASK_CNTL2_LEM) >> BITWISE_SHIFT_5);
+	printf("Tilt xp direction %ld\n", (tmp & KX022_MASK_CNTL2_RIM) >> BITWISE_SHIFT_4);
+	printf("Tilt yn direction %ld\n", (tmp & KX022_MASK_CNTL2_DOM) >> BITWISE_SHIFT_3);
+	printf("Tilt yp direction %ld\n", (tmp & KX022_MASK_CNTL2_UPM) >> BITWISE_SHIFT_2);
+	printf("Tilt zn direction %ld\n", (tmp & KX022_MASK_CNTL2_FDM) >> BITWISE_SHIFT_1);
+	printf("Tilt zp direction %ld\n", (tmp & KX022_MASK_CNTL2_FUM));
+
+	rc = kx022_read_register_value(dev, KX022_CNTL3, &tmp);
+	printf("Tilt odr %ld\n", (tmp & KX022_MASK_CNTL3_OTP) >> BITWISE_SHIFT_6);
+
+	rc = kx022_read_register_value(dev, KX022_TILT_TIMER, &tmp);
+	printf("Tilt timer %d\n", tmp);
+
+	rc = kx022_read_register_value(dev, KX022_TILT_ANGLE_LL, &tmp);
+	printf("Tilt angle ll %d\n", tmp);
+
+	rc = kx022_read_register_value(dev, KX022_TILT_ANGLE_HL, &tmp);
+	printf("Tilt angle hl %d\n", tmp);	
+}
+#endif  /*CONFIG_KX022_DIAGNOSTIC_MODE*/
+
+static void tilt_position(struct sensor_value *value)
```

Comment:
Move this function into `kx022.c`, create an `enum` for all directions and output that